### PR TITLE
New Docker Compose strategy

### LINKED ISSUE #1341: Docker support for development
This is a placeholder issue.

Once #1340 is complete, extend docker support to include development options. We want to be able to get a developer up and running with the complete OS platform. This might include things like [fake-s3](https://github.com/jubos/fake-s3) support, and options to use local data storage for the database. Also documentation for how to locally develop individual applications against the Docker stack.

- [ ] Allow developer to set locally accessible url (used in compose to redirect to correct app)

Part of #1333.

### PR DESCRIPTION
**Pull request by brew**
This PR proposes a new Docker Compose strategy for deploying the Open Spending platform. The README is mainly focused on deployment for development, though the `docker-compose.base.yaml` can be used as the basis for a production deployment too (but probably requires better documentation).

I've removed the majority of the old Dockerfiles and the /repos directory. Many of the backing services don't need their own openspending Dockerfile, and their images are referred to directly from the docker-compose.*.yml files.

I've split the main openspending services and backing services into separate compose files; docker-compose.base.yml and docker-compose.dev-services.yml respectively. The `dev-services` file is provided mainly as a convenience for development, as backing services for the production platform will be defined by resource locators to external services.

I've retained a small number of custom Dockerfiles in `/Dockerfiles` directory. These are for small services that don't have their own repository, such as Redash and FakeS3. There's also a frontend Nginx service. I've added the frontend server to this repository because it includes url location rules that apply to the whole platform, and is easier to maintain here with the docker-compose service definitions. The Nginx image is built when `docker-compose up` is run.

This PR doesn't currently address image versioning, and all `os-*` images currently use the `latest` tag.

Part of #1340 and fixes #1341. Closes #1320 without merging.

### CODE REVIEW

**Comment by akariv** [comment]
👏 👏 👏 👏 👏 
This is truly great work @brew !

I have no comments :)

**Review by akariv — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -11,8 +12,81 @@ The main use for the software is the site [openspending.org](http://openspending
...
+1. Checkout the os-conductor code from https://github.com/openspending/os-conductor into `~/src/dockerfiles/os-conductor` on your local machine.
+2. Add the following to `docker-compose.local.yml`:
+
+```yml
+version: "3"
+
+services:
+  os-conductor:
+    environment:
+      # Force python not to used cached bytecode
```
Comment (vitorbaptista) [inline README.md]: typo `used` -> `use`

Comment (vitorbaptista) [reply README.md]: Also, I'm curious: why not use the cached bytecode?

File: Dockerfiles/os-nginx-frontend/nginx.conf [hunk 2]
```diff
@@ -17,7 +16,8 @@ http {
                       '$status $body_bytes_sent "$http_referer" '
                       '"$http_user_agent" "$http_x_forwarded_for"';
 
-    access_log  /var/log/nginx/access.log  main;
+    access_log /proc/self/fd/1;
+    error_log /proc/self/fd/1 info;
```
Comment (vitorbaptista) [inline Dockerfiles/os-nginx-frontend/nginx.conf]: Nice!! This makes the logs go to the docker stdout?

File: .travis.yml [hunk 3]
```diff
@@ -1,40 +1,31 @@
...
-    - chmod +x docker-compose
-    - sudo mv docker-compose /usr/local/bin
-    - docker -v
-    - docker-compose -v
+    - sudo /etc/init.d/postgresql stop
+
 install:
-    - cd docker-config
-    - docker-compose -f build.yml build
+    - sudo mv .env.example .env
```
Comment (vitorbaptista) [inline .travis.yml]: Do we need sudo here?

File: docker-compose.base.yml [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+
+  os-packager:
+    image: 'openspending/os-packager:latest'
+    environment:
+      OS_BASE_URL:
+      OS_SNIPPETS_GA:
+      OS_SNIPPETS_RAVEN:
+      SENTRY_DSN:
+      PORT: 8000
+      WORKDIR: app
```
Comment (vitorbaptista) [inline docker-compose.base.yml]: Should it be `/app` instead?

File: docker-compose.base.yml [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+      - 8000
+
+  os-packager:
+    image: 'openspending/os-packager:latest'
+    environment:
+      OS_BASE_URL:
+      OS_SNIPPETS_GA:
+      OS_SNIPPETS_RAVEN:
+      SENTRY_DSN:
+      PORT: 8000
```
Comment (vitorbaptista) [inline docker-compose.base.yml]: I wonder, could this $PORT variable be used in the `expose`?

File: docker-compose.dev-services.yml [hunk 6]
```diff
@@ -0,0 +1,55 @@
+version: "3"
+
+services:
+
+  db:
+    image: 'postgres:10-alpine'
```
Comment (vitorbaptista) [inline docker-compose.dev-services.yml]: Are we using pgsql10 in production?

**Review by vitorbaptista — COMMENTED** [review]
Amazing work, @brew! This looks really great. I added a few comments, but they're just tiny details. Also, I think it would be good to have the commands in a Makefile, and remove the commented-out sections, but regardless of any of that, this is good to go IMHO :clap: :clap: :clap:

File: README.md (hunk 1 above)
Comment (brew) [reply README.md]: I had a few instances when working in this way where I needed to remove *.pyc files to see me changes. This isn't really a necessary part of the project, just an example of something a developer might do to configure the custom environment for their work.

File: Dockerfiles/os-nginx-frontend/nginx.conf (hunk 2 above)
Comment (brew) [reply Dockerfiles/os-nginx-frontend/nginx.conf]: Yep, we did it for our openspending.org Nginx instance as part of the DataDog work: https://hackmd.io/4gLJxSJ3TwaDvYUy-LR66g

File: docker-compose.dev-services.yml (hunk 6 above)
Comment (brew) [reply docker-compose.dev-services.yml]: No. 9.4.x. I'll change it.

**Pull request merged**